packages/taiko-client/proposer/transaction_builder/blob.go

```diff
@@ -83,31 +90,65 @@ func (b *BlobTransactionBuilder) Build(
 	signature[64] = uint8(uint(signature[64])) + 27
 
 	var (
-		to   = &b.taikoL1Address
-		data []byte
+		to            = &b.taikoL1Address
+		data          []byte
+		encodedParams []byte
+		method        string
 	)
 	if b.proverSetAddress != rpc.ZeroAddress {
 		to = &b.proverSetAddress
 	}
 
-	// ABI encode the TaikoL1.proposeBlock parameters.
-	encodedParams, err := encoding.EncodeBlockParams(&encoding.BlockParams{
-		ExtraData:      rpc.StringToBytes32(b.extraData),
-		Coinbase:       b.l2SuggestedFeeRecipient,
-		ParentMetaHash: parentMetaHash,
-		Signature:      signature,
-	})
-	if err != nil {
-		return nil, err
+	// Check if the current L2 chain is after ontake fork.
+	if !b.afterOntake {
+		blockNum, err := b.rpc.L2.BlockNumber(ctx)
+		if err != nil {
+			return nil, err
+		}
+
+		b.afterOntake = b.chainConfig.IsOntake(new(big.Int).SetUint64(blockNum))
+	}
+
+	if !b.afterOntake {
+		// ABI encode the TaikoL1.proposeBlock / ProverSet.proposeBlock parameters.
+		method = "proposeBlock"
+
+		// ABI encode the TaikoL1.proposeBlock / ProverSet.proposeBlock parameters.
+		encodedParams, err = encoding.EncodeBlockParams(&encoding.BlockParams{
+			ExtraData:      rpc.StringToBytes32(b.extraData),
+			Coinbase:       b.l2SuggestedFeeRecipient,
+			ParentMetaHash: parentMetaHash,
+			Signature:      signature,
+		})
+		if err != nil {
+			return nil, err
+		}
+	} else {
+		l1Head, err := b.rpc.L1.HeaderByNumber(ctx, nil)
+		if err != nil {
+			return nil, err
+		}
+		// ABI encode the TaikoL1.proposeBlockV2 / ProverSet.proposeBlockV2 parameters.
+		method = "proposeBlockV2"
+
+		if encodedParams, err = encoding.EncodeBlockParamsOntake(&encoding.BlockParams2{
+			Coinbase:       b.l2SuggestedFeeRecipient,
+			ExtraData:      rpc.StringToBytes32(b.extraData),
+			ParentMetaHash: parentMetaHash,
+			AnchorBlockId:  l1Head.Number.Uint64(),
+			Timestamp:      l1Head.Time,
+		}); err != nil {
+			return nil, err
+		}
 	}
 
 	if b.proverSetAddress != rpc.ZeroAddress {
-		data, err = encoding.ProverSetABI.Pack("proposeBlock", encodedParams, []byte{})
+		data, err = encoding.ProverSetABI.Pack(method, encodedParams, []byte{})
```

Comment:
The value of the second parameter means that `_txList` is empty. Is this expected?

Reply:
Yes, as before.